internal/security/credentials.go

```diff
@@ -40,3 +40,44 @@ func (s *SecretProvider) GetDatabaseCredentials(database db.DatabaseInfo) (commo
 		Password: secrets["password"],
 	}, nil
 }
+
+// GetSecrets retrieves secrets from a secret store.
+// path specifies the type or location of the secrets to retrieve.
+// keys specifies the secrets which to retrieve. If no keys are provided then all the keys associated with the
+// specified path will be returned.
+func (s *SecretProvider) GetSecrets(path string, keys ...string) (map[string]string, error) {
+
+	// check cache for keys
+	allKeysExistsInCache := true
+	cachedSecrets := s.secrets[path]
```

Comment:
this call can be improved and avoid going to the for loop if you 
`cachedSecrets := s.secrets[path]` -> `cachedSecrets, cacheExists := s.secrets[path]` including a new var cacheExists and testing the condition 'cacheExists' in  an if statement around all of the code from line 53 to line 67.

Reply:
+1 for using cacheExists to check the existence in the map.

Reply:
Once Put is implements, it will need to clear the cache, so probably what the caching to separated some how.

Reply:
good call